### DISCUSSION TITLE
after training on VoxCeleb1+2, How to do inference?

### DISCUSSION THREAD

**Discussion by dragen1860** [speechbrain/speechbrain#1644] [Q&A]
Dear all:
  I succeed to train on VoxCeleb1+2 dataset:
```python
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3371/3371 [08:50<00:00,  6.36it/s]
speechbrain.nnet.schedulers - Changing lr from 0.00076 to 0.00072
speechbrain.utils.train_logger - epoch: 6, lr: 7.63e-04 - train loss: 2.98e+02 - valid loss: 18.50, **valid ErrorRate: 3.86**
speechbrain.utils.checkpoints - Saved an end-of-epoch checkpoint in results/xvect_augment/1987/save/CKPT+2021-09-11+08-48-12+00
speechbrain.utils.checkpoints - Deleted checkpoint in results/xvect_augment/1987/save/CKPT+2021-09-11+02-35-36+00
speechbrain.utils.checkpoints - Deleted checkpoint in results/xvect_augment/1987/save/CKPT+2021-09-11+08-36-01+00

```

But when I try to do inference on wav pairs, I feel confused on how to create modules and load checkpoint from `hyperparams.yaml`.

I know how to do inference from Huggingface hub as follows:
```python
            self.speechbrain = EncoderClassifier.from_hparams(
                source="speechbrain/spkrec-xvect-voxceleb",
                savedir="pretrained_models/spkrec-xvect-voxceleb")

```

Anyone give some advice on how to loaded trained checkpoint and do inference? thank you very much.

**Comment by rohaantahir** [comment]
@mravanelli hi could you please also tell how mean_var_norm_emp.ckpt is generated using voxceleb recipe. I traned the model and run all the three python commands available in the redme file. Could you please tell how it can be generated without mean_var_norm_emp.ckpt the results is not authentic it always show False with similarity score 0.

**Comment by rohaantahir** [comment]
@dragen1860 could you also please check at your end that the above mentioned file is generated or not ?

**Comment by dragen1860** [comment]
@rohaantahir Thanks for your reply. I found the training procedure is extremely slow. When i trained on voxceleb1+2, it will takes me 25 hours for single epoch. I do not finished the total training yet. But there is no `mean_var_norm_emp.ckpt` generated in checkpoint directory.

**Comment by rohaantahir** [comment]
There is a file named with nomalizer.ckpt available maybe this would be same but I am not sure about it.

**Comment by dragen1860** [comment]
@rohaantahir Thank for reply. How many hours you spent on training on single epoch? and how many gpu cards you used? thank you.

**Comment by pnsafari** [comment]
Hi @rohaantahir @dragen1860, I have trained on only vox2. But for the `plda` verification I see [this line](https://github.com/speechbrain/speechbrain/blob/13798b9576779b82f70c8f65dc0e166da5c09c10/recipes/VoxCeleb/SpeakerRec/speaker_verification_plda.py#L241) in `speaker_verification_plda.py` file. Now I have two questions:

1. Only the `dev` set of vox1 is used which I cannot understand why?
2. Why this `dev` set is again divided into two 90%-10% train and test?

Am I missing something or what?

**Comment by dragen1860** [comment]
@pnsafari 1. THE community usually use vox1 dev set for testing. it's a custom. you can use vox2 dev set for testing as well. no special reasons.     2. 90~10 traiinging and testing is one of the most common setting.
